src/dom_components/model/Component.ts

```diff
@@ -897,15 +906,24 @@ export default class Component extends StyleableModel<ComponentProperties> {
     this.off(event, this.initTraits);
     this.__loadTraits();
     const attrs = { ...this.get('attributes') };
+    const traitDataVariableAttr: ObjectAny = {};
     const traits = this.traits;
     traits.each((trait) => {
-      if (!trait.changeProp) {
-        const name = trait.getName();
-        const value = trait.getInitValue();
+      const name = trait.getName();
+      const value = trait.getInitValue();
+
+      if (trait.changeProp) {
+        this.set(name, value);
```

Comment:
you know I'm not sure about this... mainly because with `changeProp` I'd expect the trait to get the value from the component. This probably would make sense when the property of the component is `undefined`.
wdyt?

Reply:
Please see my reference in the test case.

Reply:
makes sense, let's just ensure the value is not undefined

Reply:
* https://github.com/GrapesJS/grapesjs/pull/6101